test/scripts/gh-actions/setup-poetry.sh

```diff
@@ -20,7 +20,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-export POETRY_VERSION=1.4.0
+export POETRY_VERSION=1.7.1
```

Comment:
@terrytangyuan We already have a PR for [upgrading poetry](https://github.com/kserve/kserve/pull/3271). I think it is better not to include it here. WDYT ?

Reply:
I had issue locally with POETRY_VERSION=1.4.0 for some reason on my Mac. This is the only workaround I found to run `./hack/python-release.sh` successfully.